### DISCUSSION TITLE
How to use another illuminant than D65 with JzAzBz colour model?

### DISCUSSION THREAD

**Discussion by waybackrider** [colour-science/colour#831] [Q&A]
Hi,
Im trying to use CAM16UCS Space for Color Difference Calculations as i learned this is more suitable for near Colors than Lab.
In my case the XYZ Tristimulus Values are from an emissive Measurement (a LED) and BruceLindbloom says use Illuminant E for this (Note in: http://brucelindbloom.com/index.html?SpectCalcSpreadsheets.html)
So why is XYZ_to_JzAzBz fixed to D65, maybe there is another way?

Thanks!

**Comment by thomasmansencal** [comment]
Hi @waybackrider,

As you saw, there is no parameterization for the illuminant for the *JzAzBz* colour model, the typical assumption is that colours are chromatically [adapted](https://cie.co.at/eilvterm/17-23-081) to D65 but not necessarily: D65 is actually the [adopted whitepoint](https://cie.co.at/eilvterm/17-32-033), colours lying under its axis are achromatic.

This would be important for example if you had a bunch of reflectances that had been computed under another illuminant, e.g. tungsten: as you might want to keep whites neutral and achromatic, you would need to chromatically adapt from tungsten to D65.

In your case, and considering that the stimulus is self-luminous, I would not perform any chromatic adaptation: As you are comparing irradiance values, you could think as if the observer was adapting to each one in turn.

Cheers,

Thomas

PS: We have all the tools to perform the spectral integration to XYZ, see `colour.sd_to_XYZ` definition, it is also possible to perform the entire conversion in one go with `colour.convert` definition.

**Reply by waybackrider** [reply]
Hi,
thanks for your Response!

I don't understand what you mean by "As you are comparing irradiance values". I'm trying to compare XYZ Values, until now using CIE2000, now hopefully using JzAzBz for better Results.
Furthermore i'm not dealing with spectral data, just using a [Colorimeter](https://ams.com/as73211) getting the XYZ.

Maybe it is hopeful to shortly describe my overall goal, maybe you can give me some hints if i am  going in the wrong direction, i'm juster entering the World of color science, so i'm really thankful for you help and of course for your great lib!

I am trying to group an List of Leds in Packs of 5 which look most equally. 
Later i learned there are other and better Tools out there for this Task, i just wanted to give it a try anyway and digging deeper into Colorscience.
So maybe another Question i am coping with is about normalizing XYZ Tristimulus Values. Depending on the Colorimeter Settings they get really huge (e.g [8700, 9468, 10881]. Now i am normalizing by dividing by sum and getting to xyY Space, but Y is very huge compared to x and y, can this be true?

Grettings!

**Reply by thomasmansencal** [reply]
Hello,

Allright, thanks for the details, it helps! The CIE XYZ tristimulus values you get from the colorimeter are illuminant independent, i.e. as if they were measured with an equal energy illuminant, i.e. CIE Illuminant E. You should be able to push them directly into `colour.XYZ_to_JzAzBz` definition.

The DAC of your colorimeter is 16-bit by the look of it, thus you need to divide your array by `2^16 -1` and it is indeed measuring Irradiance:

> With its irradiance responsivity factor and conversion time, sensor board supports a dynamic range of
1-to-250,000,000 and achieves an accuracy of up to 16-bit (internal 24-bit) signal resolution with an
irradiance responsivity per count down to 0.0005 pW/cm².

**Reply by waybackrider** [reply]
Hi,

i just tried you suggestion of dividing by 2^16-1 and i can't believe it's so simple, you are a Hero!
The plotted Color is now really close to what i'm seeing! :-)

Is the ability of measuring Irradiance useful for my goal?
I'm asking because because i also want to try with white Leds and i came across this is better done by utilizing CCT than DeltaE, can you confirm this?

Once again, many Thanks!

**Reply by thomasmansencal** [reply]
Irradiance here is just the type of spectrum you are measuring, typically the spectrum of emission source. CCT will not give you a measure of how two colours are different, this must be done in a perceptually uniform space by measuring the distances between two colour samples.

**Reply by waybackrider** [reply]
Hi,
my Results are getting better and better, two more Questions i came up with:

- Using "delta_E_CAM16UCS" with Jzazb computes much smaller Deltas than CIE2000, about Factor 1000 smaller. Can this be true? Isn't the Meaning of this Delta standardizied in some way?

- There are two other possible Methods "delta_E_CAM16LCD" and "delta_E_CAM16SCD". I can't find deeper Information about this LCD and SCD, except ist meaning, Large and Small Color Difference. As my Measurements are pretty close together, is it useful to utilize "delta_E_CAM16SCD"? Whats behind SCD?

Feel free to move this to an new Discussion if you find so!

Thanks!
